Common/Wrapper/TraktManager.swift

```diff
@@ -216,6 +238,156 @@ public class TraktManager {
         return try JSONSerialization.data(withJSONObject: json, options: [])
     }
     
+    // MARK: - Authenticate Devices
+    
+    public func getDeviceCode(completionHandler: @escaping DataResultCompletionHandler) throws {
+        guard let clientID = clientID
+            else {
+                completionHandler(.error(error: nil))
+                return
+        }
+        
+        
+        let urlString = "https://\(baseURL!)/oauth/device/code"
+        let url = URL(string: urlString)
+        guard var request = mutableRequestForURL(url, authorization: false, HTTPMethod: .POST) else {
+            completionHandler(.error(error: nil))
+            return
+        }
+        
+        let json = [
+            "client_id": clientID,
+        ]
+        request.httpBody = try JSONSerialization.data(withJSONObject: json, options: [])
+        session._dataTask(with: request) { [weak self] (data, response, error) -> Void in
+            guard let welf = self else { return }
+            guard error == nil else {
+                completionHandler(.error(error: error))
+                return
+            }
+            
+            // Check response
+            guard let HTTPResponse = response as? HTTPURLResponse,
+                HTTPResponse.statusCode == StatusCodes.Success
+                else {
+                    if let HTTPResponse = response as? HTTPURLResponse {
+                        completionHandler(.error(error: welf.createErrorWithStatusCode(HTTPResponse.statusCode)))
+                    } else {
+                        completionHandler(.error(error: nil))
+                    }
+                    return
+            }
+            // Check data
+            guard
+                let data = data else {
+                    completionHandler(.error(error: nil))
+                    return
+            }
+            
+            do {
+                if let deviceCodeDict = try JSONSerialization.jsonObject(with: data, options: []) as? [String: AnyObject] {
+                    
+                    welf.deviceCode = deviceCodeDict["device_code"] as? String
+                    welf.userCode = deviceCodeDict["user_code"] as? String
+                    welf.verificationURL = deviceCodeDict["verification_url"] as? String
+                    welf.timeInterval = deviceCodeDict["interval"] as? TimeInterval
+                    welf.expiresIn = deviceCodeDict["expires_in"] as? TimeInterval
```

Comment:
I think decoding into an object would be preferable to this old way of decoding JSON